### PR TITLE
FileHandle: Mbed OS 5.12 updates

### PR DESCRIPTION
**Pull request by kjbracey**
Cover changes in https://github.com/ARMmbed/mbed-os/pull/9797

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

File: docs/api/platform/FileHandle.md [hunk 1]
```diff
@@ -112,6 +117,14 @@ Important notes on sigio:
 
 Ordinary files do not generate sigio callbacks because they are always readable and writable.
 
+#### Suspending a device
+
+Having a device open through a `FileHandle` may cost power, especially if open for input. For example, for `UARTSerial` to be able to receive data, the system must not enter deep sleep, so deep sleep is prevented while the `UARTSerial` is active.
+
+To permit power saving, you can close or destroy the device, or you can indicate that you do not currently require input or output by calling `FileHandle::enable_input` or `FileHandle::enable_output`. Disabling input or output effectively suspends the device in that direction, which can permit power saving.
```
Comment (AnotherButler) [inline docs/api/platform/FileHandle.md]: What do we mean by destroy the device to permit power saving? This sounds counterintuitive to me.

**Review by AnotherButler — COMMENTED** [review]
Thanks for the PR. I've left one query.

File: docs/api/platform/FileHandle.md (hunk 1 above)
Comment (kjbracey) [reply docs/api/platform/FileHandle.md]: That should be close or destroy the `FileHandle` (by which you're accessing the device, which would then permit the device to shut down).

**Review by AnotherButler — APPROVED** [review]
Nice work on this 👍 Thanks

**Pull request merged**